### DISCUSSION TITLE
Multiple brushes & non-linear brush

### DISCUSSION THREAD

**Discussion by mlisthenewcool** [airbnb/visx#1163] [Q&A]
Is there any reference for the following use cases ?

## Multiple brushes

Hold ctrl or alt or shift + selection for :

![multiple-selections](https://user-images.githubusercontent.com/43213629/114559041-9d559f00-9c6b-11eb-8196-43fdebf5e908.png)

## Non-linear brush

![non-linear-selection](https://user-images.githubusercontent.com/43213629/114559047-9dee3580-9c6b-11eb-9428-b14b8d867fdc.png)

**Comment by williaster** [comment]
Hey @mlisthenewcool 👋 , interesting use cases! I haven't seen these discussed before in the `visx` community, but they're clearly valid / common vis use-cases. 

I think for the first case (multiple brushes), you could probably implement it already using the existing `Brush` component. You would need to handle the `ctrl` key listeners yourself, but then there's nothing stopping you from rendering multiple brushes and keeping track of their selections.

For the second case (non-linear), we could probably build something like that pretty easily on top of `@visx/drag` since it allows you to draw arbitrary shapes (see this [demo](https://airbnb.io/visx/drag-ii)). Building that as a re-usable new component in `@visx/brush` would be pretty great so others don't have to manage all of the complexity.
